src/diffusers/models/transformers/transformer_sd3.py

```diff
@@ -67,43 +69,42 @@ def __init__(
         joint_attention_dim: int = 4096,
         caption_projection_dim: int = 1152,
         pooled_projection_dim: int = 2048,
-        out_channels: int = 16,
+        out_channels: Optional[int] = 16,
         pos_embed_max_size: int = 96,
     ):
         super().__init__()
-        default_out_channels = in_channels
-        self.out_channels = out_channels if out_channels is not None else default_out_channels
-        self.inner_dim = self.config.num_attention_heads * self.config.attention_head_dim
+        self.out_channels = out_channels or in_channels
+        self.inner_dim = num_attention_heads * attention_head_dim
 
         self.pos_embed = PatchEmbed(
-            height=self.config.sample_size,
```

Comment:
I think whether to use the config object inside models is more of a collective decision. We have it set up like this in a few places. Functionally, they are equivalent but we just need to commit to an approach. My personal preference is to use the variable directly as done here. cc: @yiyixuxu  @sayakpaul

Reply:
Yeah using a variable directly is what I have been following recently as well. During the `forward()` if the variable is accessible through `self.config` that takes priority, but if not (such as `inner_dim`) just assign then to `self` during `__init__()`.